core/arch/arm/kernel/abort.c

```diff
@@ -108,13 +108,45 @@ static __maybe_unused const char *abort_type_to_str(uint32_t abort_type)
 	return "undef";
 }
 
+#ifdef ARM64
+static __maybe_unused const char *fault_to_str(uint32_t esr)
+{
+	switch ((esr >> ESR_EC_SHIFT) & ESR_EC_MASK) {
+	case ESR_EC_IABT_EL0:
+	case ESR_EC_IABT_EL1:
+	case ESR_EC_DABT_EL0:
+	case ESR_EC_DABT_EL1:
+		switch (esr & ESR_FSC_MASK) {
+		case ESR_FSC_ALIGN:
+			return " (alignment fault)";
+		case ESR_FSC_TRANS_L0:
+		case ESR_FSC_TRANS_L1:
+		case ESR_FSC_TRANS_L2:
+		case ESR_FSC_TRANS_L3:
+			return " (translation fault)";
+		default:
+			break;
+		}
+	default:
+		break;
+	}
+	return "";
+}
+#else
+static __maybe_unused const char *fault_to_str(uint32_t fault __unused)
+{
+	return "";
```

Comment:
The problem is `core_mmu_lpae.c` does not handle exceptions from EL0. Do you think it is safe to do the following?
```
--- a/core/arch/arm/mm/core_mmu_lpae.c
+++ b/core/arch/arm/mm/core_mmu_lpae.c
@@ -813,6 +813,8 @@ enum core_mmu_fault core_mmu_get_fault_type(uint32_t fault_descr)
        case ESR_EC_SP_ALIGN:
        case ESR_EC_PC_ALIGN:
                return CORE_MMU_FAULT_ALIGNMENT;
+       case ESR_EC_IABT_EL0:
+       case ESR_EC_DABT_EL0:
        case ESR_EC_IABT_EL1:
        case ESR_EC_DABT_EL1:
                switch (fault_descr & ESR_FSC_MASK) {
```
I can try it anyway.